src/cmap/wire_protocol/on_data.ts

```diff
@@ -0,0 +1,99 @@
+import { type EventEmitter } from 'events';
+
+import { List, promiseWithResolvers } from '../../utils';
+
+type PendingPromises = Omit<
+  ReturnType<typeof promiseWithResolvers<IteratorResult<Buffer>>>,
+  'promise'
+>;
+
+export function onData(emitter: EventEmitter, options: { signal: AbortSignal }) {
```

Comment:
All of in the virtual office know the reasoning for this but maybe we should still add some comments here on why we implemented this.